src/libs/ajax/DatasetBuilder.ts

```diff
@@ -1,11 +1,93 @@
+// Types that can be used to create a criteria.
+
+export interface DomainType {
+  id: number;
+  category: string;
+  values: string[];
+}
+
+type DataType = 'range' | 'list';
+
+export interface ProgramDataType {
```

Comment:
in a similar vein of the other type name comment, feels like this should be called ProgramData (and similar naming tweaks for the sub-classes), and DataType on line 9 should be called ProgramDataType since it's an enum-like type.

Reply:
The problem is that this isn't program data. It's the type of program data that will be searched over. For example, the program data for Gender might be Male, but the concept this type represents is Gender, which includes all of the possible values of the program data. The name could be changed, but the concept it represents is the _type_ of program data, so it should have a name that reflect this concept somehow.

Reply:
maybe ProgramDataFilter?

I kinda cringe when "Type" is used since it is so easy for this to be misunderstood as a broader naming convention of naming all types that way.

Reply:
> maybe ProgramDataFilter?

In this world, the criteria are what's actually filtered on. This is the object that's used as a source to generate a criteria. Looking in the thesaurus I see `Kind` `Model` `Exemplar`, etc

> I kinda cringe when "Type" is used since it is so easy for this to be misunderstood as a broader naming convention of naming all types that way.

I agree! I don't think `Filter` is correct but there may be some other term that could work.

Reply:
@s-rubenstein and I talked it over and are going with `Option` for this concept for now. Maybe later on we'll have an idea for something better.